sale_automatic_workflow/models/sale_workflow_process.py

```diff
@@ -35,36 +35,41 @@ def _default_filter(self, xmlid):
         default='direct',
     )
     validate_order = fields.Boolean(string='Validate Order')
-    order_filter_domain = fields.Char(
+    order_filter_domain = fields.Text(
         string='Order Filter Domain',
-        default="[('state', '=', 'draft')]"
+        related='order_filter_id.domain',
```

Comment:
@lasley I think @lmignon would say that generally we set related fields as readonly.
This to avoid modification through this related field.

Moreover, these related fields are 'informative' field => if we want to modify it, we do it through the Many2one.

The behaviour you show is the 'char_domain' one (ok it's weird to have sale.order readonly as we set it on ir.filter.domain...).

So, no restriction in usage with readonly (as we modify through the other field - Many2one)